### PR TITLE
refactor filter generators

### PR DESCRIPTION
**Pull request by thanhlm91**
1. Refactor  class 'Repository\ComposableFilterRepository.cs', rename it to  'FilterParameters' & put into folder '/Generators/Filters'
2. Create a base class named 'BaseFilterGenerator' in folder '/Generators/Filters/Base'
3. Refactor & rename the following classes & put them into folder '/Generators/Filters' 
   - 'Filters\ComposableFilterByCodeGenerator.cs' => 'ComposableFilterGenerator'
   - 'Filters\FilterByCodeGenerator.cs' -> 'FilterGenerator'
4. Put logic of generating RowPermissions inside class 'ComposableFilterGenerator' as 'RowPermissionsReadInfo' extends 'ComposableFilterByInfo'

### CODE REVIEW

File: RhetosPackages/Source/Angular2ModelGenerator/Generators/Filters/Base/BaseFilterGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Angular2ModelGenerator.Templates;
+using Rhetos.Compiler;
+using Rhetos.Dsl;
+using Rhetos.Dsl.DefaultConcepts;
+
+namespace Angular2ModelGenerator.Generators.Filters.Base
+{
+    public abstract class BaseFilterGenerator<T> : IFilterGenerator
+    {
+        public abstract bool IsComposable { get; }
```
Comment (ngoquoc) [inline RhetosPackages/Source/Angular2ModelGenerator/Generators/Filters/Base/BaseFilterGenerator.cs]: I think Composable is too specific to place in base class of filters. There might be some other types of filters which we can't anticipate and we would have to modify this base class once new type of filter is introduced.

**Comment by thanhlm91** [comment]
This PR will resolve issue #1

File: RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.csproj [hunk 2]
```diff
@@ -116,11 +119,9 @@
     <Compile Include="Property\PropertyCodeGeneratorHelper.cs" />
     <Compile Include="Property\ReferencePropertyCodeGenerator.cs" />
     <Compile Include="Property\DecimalPropertyCodeGenerator.cs" />
-    <Compile Include="Repository\ComposableFilterRepository.cs" />
     <Compile Include="SimpleBusinessLogic\MaxLengthTagCodeGenerator.cs" />
     <Compile Include="SimpleBusinessLogic\MinLengthTagCodeGenerator.cs" />
     <Compile Include="SimpleBusinessLogic\RegExTagCodeGenerator.cs" />
-    <Compile Include="SimpleBusinessLogic\RowPermissionsCodeGenerator.cs" />
```
Comment (bantolov) [inline RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.csproj:130]: RowPermissionsCodeGenerator.cs is removed from the project, but the file is not deleted from disk. Please check if this file should be removed or added back to the project.

Comment (thanhlm91) [reply RhetosPackages/Source/Angular2ModelGenerator/Angular2ModelGenerator.csproj:130]: I deleted that file.

**Pull request merged**